### PR TITLE
Support signal_t refactor, add platform-specific code support

### PR DESCRIPTION
**Pull request by jacobmcnamee**
Adds defines and indexing functions enabling platform-specific support of individual code types.
```
/* Platform-specific code support ==== */
#define CODE_GPS_L1CA_SUPPORT     1
#define CODE_GPS_L2CM_SUPPORT     0
#define CODE_SBAS_L1CA_SUPPORT    1
/* =================================== */
```
Unsupported codes are excluded from signal count defines and indexing functions.

Support for swift-nav/libswiftnav#303.

### CODE REVIEW

**Comment by jacobmcnamee** [comment]
Requires update to libswiftnav pending PR.

File: .gitmodules [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	url = https://github.com/libopencm3/libopencm3.git
 [submodule "libswiftnav"]
 	path = libswiftnav
-	url = https://github.com/swift-nav/libswiftnav.git
+	url = https://github.com/jacobmcnamee/libswiftnav.git
```
Comment (pmiettinen) [inline .gitmodules]: You probably don't mean to change this submodule url?

Comment (jacobmcnamee) [inline .gitmodules]: No, it's just for testing until the libswiftnav change is merged.

File: src/signal.c [hunk 2]
```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2011-2016 Swift Navigation Inc.
```
Comment (pmiettinen) [inline src/signal.c]: Should be only '2016'?

File: src/signal.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SWIFTNAV_SIGNAL_H
+#define SWIFTNAV_SIGNAL_H
+
+#include <libswiftnav/common.h>
+#include <libswiftnav/signal.h>
+
+/* Platform-specific code support ==== */
```
Comment (pmiettinen) [inline src/signal.h]: Is the plan to ultimately set these defines in build system?

Comment (jacobmcnamee) [inline src/signal.h]: Yes, definitely. Either makefile or board.h. Just waiting for other changes to set this up.

**Pull request merged**